scripts/buildWeeklyRequestCountDocuments/index.js

```diff
@@ -0,0 +1,53 @@
+const admin = require('firebase-admin')
+const { initFirebaseAdmin } = require('../utils/initFirebaseAdmin.js')
+const { firebaseTimestampToMoment } = require('../../mixins/date/firebaseTimestampToMoment')
+const { getFormattedEndOfWeek } = require('../../mixins/date/getFormattedEndOfWeek')
+const { getFormattedStartOfWeek } = require('../../mixins/date/getFormattedStartOfWeek')
+
+initFirebaseAdmin()
+
+const buildRequestCountDocuments = async () => {
+  const weeklyRequestCreation = {}
+
+  try {
+    const allRequests = admin
+      .firestore()
+      .collectionGroup('requests')
+
+    const snapshots = await allRequests.get()
+
+    await Promise.all(
+      snapshots.docs.map(async (doc) => {
+        console.log(`Adding request: ${doc.id} to the count`)
+
+        const data = doc.data()
+        const startDate = `${getFormattedStartOfWeek(firebaseTimestampToMoment(data.createdAt))}`
+        const endDate = `${getFormattedEndOfWeek(firebaseTimestampToMoment(data.createdAt))}`
+        const mapKeyForRequestDate = `${startDate}-${endDate}`
+
+        if (weeklyRequestCreation[mapKeyForRequestDate]) {
+          weeklyRequestCreation[mapKeyForRequestDate].amountOfRequests++
+        } else {
+          weeklyRequestCreation[mapKeyForRequestDate] = {
+            startDate,
+            endDate,
+            amountOfRequests: 1
+          }
+        }
+      })
+    )
+
+    Object.entries(weeklyRequestCreation).forEach(async ([key, data]) => {
+      await admin
+        .firestore()
```

Comment:
You're not `await`ing the operation at the main level. Change `forEach` with `map` and surround it with a `await Promise.all(...)`